src/fastfield/serializer/mod.rs

```diff
@@ -142,17 +143,19 @@ impl CompositeFastFieldSerializer {
             base_value: u64,
             max_value: u64,
             num_vals: u64,
-            gcd: u64,
+            gcd: DividerU64,
         }
+
         impl<T: FastFieldDataAccess> FastFieldDataAccess for GCDWrappedFFAccess<T> {
             fn get_val(&self, position: u64) -> u64 {
-                (self.fastfield_accessor.get_val(position) - self.base_value) / self.gcd
+                self.gcd
+                    .divide(self.fastfield_accessor.get_val(position) - self.base_value)
             }
-            fn iter<'b>(&'b self) -> Box<dyn Iterator<Item = u64> + 'b> {
+            fn iter(&self) -> Box<dyn Iterator<Item = u64> + '_> {
                 Box::new(
                     self.fastfield_accessor
                         .iter()
-                        .map(|val| (val - self.base_value) / self.gcd),
+                        .map(|val| self.gcd.divide(val - self.base_value)),
```

Comment:
if you release a new version with the `Div` impl, the code could stay the same, which is more readable